File: src/net/dodian/uber/game/model/player/casino/Symbol.java
package net.dodian.uber.game.model.player.casino;

public class Symbol {
  int id;
  int[] triggers;
  String symbol = "", color = "";

  public Symbol(int id, String symbol, int[] triggers) {
    this.id = id;
    this.symbol = symbol;
    this.triggers = triggers;
  }

  public boolean check(int stop) {
    for (int i : triggers) {
      if (i == stop)
        return true;
    }
    return false;
  }

  public String getSymbol() {
    return symbol;
  }

  public int getId() {
    return id;
  }

  public void setSymbol(String symbol) {
    this.symbol = symbol;
  }

  public void setColor(String color) {
    this.color = "@" + color + "@";
  }

  public String getColor() {
    return color;
  }

  public String output() {
    return color + symbol;
  }
}

File: game-server/src/main/kotlin/net/dodian/uber/game/model/player/casino/Symbol.kt
package net.dodian.uber.game.model.player.casino

class Symbol(
    private val id: Int,
    private var symbol: String,
    private val triggers: IntArray,
) {
    private var color: String = ""

    fun check(stop: Int): Boolean {
        for (i in triggers) {
            if (i == stop) {
                return true
            }
        }
        return false
    }

    fun getSymbol(): String = symbol

    fun getId(): Int = id

    fun setSymbol(symbol: String) {
        this.symbol = symbol
    }

    fun setColor(color: String) {
        this.color = "@$color@"
    }

    fun getColor(): String = color

    fun output(): String = color + symbol
}
